src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9987,6 +9987,9 @@ uint32 createProcHitMask(SpellNonMeleeDamage* damageInfo, SpellMissInfo missCond
 
 void Unit::ProcSkillsAndReactives(bool isVictim, Unit* procTarget, uint32 typeMask, uint32 hitMask, WeaponAttackType attType)
 {
+    if (!procTarget)
```

Comment:
this change does nothing, possibly only enables code analysis warnings about it being checked twice and always true later in the function

Reply:
it checks if procTarget pointer is not null so that the code below won't cause a segmentation fault.

Reply:
As I said, this is already checked 5 lines below

Reply:
I see it now, somehow missed it